tests/__init__.py

```diff
@@ -192,6 +193,17 @@ def create_auth_provider(server, account_handler, config=None):
     return LdapAuthProvider(config, account_handler=account_handler)
 
 
+def make_awaitable(result: Any) -> Awaitable[Any]:
+    """
+    Makes an awaitable, suitable for mocking an `async` function.
+    This uses Futures as they can be awaited multiple times so can be returned
+    to multiple callers.
```

Comment:
Looks like this would be a nice change to put into synapse so that we don't have to use side_effect + `lambda` in a lot of places.

Reply:
See matrix-org/synapse#8261